src/viewer/viewer.ts

```diff
@@ -1032,74 +1025,78 @@ export default class Viewer {
   }
 
+  /**
+   * Convert an absolute clip value to a relative one using bRadius.
+   *
+   * 0.0 -> 50.0
+   * bRadius -> 0.0
+   */
+  absoluteToRelative (d: number) :number {
+    return 50 * (1 - d / this.bRadius)
+  }
+
+  /**
+   * Convert a relative clip value to an absolute one using bRadius
+   *
+   * 0.0 -> bRadius
+   * 50.0 -> 0.0
+   */
+  relativeToAbsolute (d: number) : number {
+    return this.bRadius * (1 - d/50)
+  }
+
+  /**
+   * Intepret clipMode, clipScale and set the camera and fog clipping.
+   * Also ensures bRadius and cDist are valid
+   */
   private __updateClipping () {
     const p = this.parameters
 
-    const debugClipping = false
+    // bRadius must always be updated for material-based clipping
+    // and for focus calculations
+    this.bRadius = Math.max(10, this.boundingBoxLength * 0.5)
 
-    // clipping
+    // FL: Removed below, but leaving commented as I don't understand intention
+    // this.bRadius += this.boundingBox.getCenter(this.distVector).length()
+
+    if (!isFinite(this.bRadius)) {
+      this.bRadius = 50
+    }
 
-    // cDist = distVector.copy( camera.position )
-    //           .sub( controls.target ).length();
     this.cDist = this.distVector.copy(this.camera.position).length()
-    if (debugClipping)
-      console.log( "cDist", this.cDist )
     if (!this.cDist) {
       // recover from a broken (NaN) camera position
       this.camera.position.set(0, 0, p.cameraZ)
       this.cDist = Math.abs(p.cameraZ)
     }
 
-    if (p.clipScale !== 'absolute' || p.fogScale !== 'absolute') {
-      // Compute bbox radius if needed for relative clip/fog
-      this.bRadius = Math.max(10, this.boundingBoxLength * 0.5)
-      this.bRadius += this.boundingBox.getCenter(this.distVector).length()
-      // console.log( "bRadius", this.bRadius )
-      if (this.bRadius === Infinity || this.bRadius === -Infinity || isNaN(this.bRadius)) {
-        // console.warn( "something wrong with bRadius" );
-        this.bRadius = 50
-      }
-    }
+    // fog
+    const fog = this.scene.fog as Fog
+    fog.color.set(p.fogColor)
+
+    if (p.clipMode === 'camera') {
+      // Always interpret clipScale as absolute for clipMode camera
 
-    if (p.clipMode == 'camera') {
-      // clip camera mode ignores clipScale; always absolute
       this.camera.near = p.clipNear
       this.camera.far = p.clipFar
-    }
-    else {
-      // scene mode
-      if (p.clipScale == 'absolute') {
-        // absolute scene mode: offset clip planes from scene center
-        // (note: positive clipNear means closer to the camera)
+      fog.near = p.fogNear
+      fog.far = p.fogFar
+
+    } else {
+
+      if (p.clipScale === 'absolute') {
+
         this.camera.near = this.cDist - p.clipNear
         this.camera.far = this.cDist + p.clipFar
+        fog.near = this.cDist - p.fogNear
+        fog.far = this.cDist + p.fogFar
+
       } else {
-        // relative scene mode (default): convert percentages to Angstroms
```

Comment:
I liked those comments, but OK

Reply:
Sorry - they can go back! (It was the result of several refactors rather than a conscious decision to remove them!)

Reply:
They're back :)